src/PBL_example.jl

```diff
@@ -0,0 +1,239 @@
+    wₘ = IfElse.ifelse((stable >= 0), u_star/ϕₕ, i2)
+    return wₘ
+end
+
+"""
+Function to calculate wₜ
+"""
+function calc_wₜ(z, h, u_star, ϕₕ, wₘ, Pr)
+    wₜ = IfElse.ifelse((z/h <= 0.1), u_star/ϕₕ, wₘ/Pr)
+end
+
+"""
+Function to calculate w_C_0
+"""
+function calc_w_C_0(C0, C1, z1, u_star, θᵥ_0, w_θv_0)
+    L = calc_L(u_star, θᵥ_0, w_θv_0)
+    ϕₕ = calc_ϕₕ(-1,z1,L)
+    w_C_0 = -k*u_star*z1/ϕₕ*((C1-C0)/z1)
+    return w_C_0
+end
+
+"""
+Function to calculate Pr
+"""
+function calc_Pr(w_star, u_star)
+    Pr = IfElse.ifelse((w_star/u_star <= 10), 1-(w_star/u_star)/10*0.4, 0.6)
+end
+
+"""
+Function to calculate Kc
+"""
+function calc_Kc(w_u_0, w_v_0, θᵥ_0, w_θv_0, θᵥ_1, z, h)
+    u_star = calc_u_star(w_u_0, w_v_0)
+    L = calc_L(u_star, θᵥ_0, w_θv_0)
+    stable = calc_stable(θᵥ_0,θᵥ_1)
+    ϕₕ = calc_ϕₕ(stable, z, L)
+    w_star = calc_w_star(θᵥ_0, w_θv_0, h)
+    wₘ = calc_wₘ(stable, z, h, u_star, ϕₕ, w_star)
+    Pr = calc_Pr(w_star, u_star)
+    wₜ = calc_wₜ(z, h, u_star, ϕₕ, wₘ, Pr)
+    Kc = k*wₜ*z*(1-z/h)^2
+    return Kc
+end
+
+"""
+function to calcualte γc
+"""
+function calc_γc(θᵥ_0, θᵥ_1, z, h, w_θv_0, w_u_0, w_v_0, C0, C1, z1)
+    stable = calc_stable(θᵥ_0,θᵥ_1)
+    w_star = calc_w_star(θᵥ_0, w_θv_0, h)
+    u_star = calc_u_star(w_u_0, w_v_0)
+    w_C_0 = calc_w_C_0(C0, C1, z1, u_star, θᵥ_0, w_θv_0)
+    ϕₕ = calc_ϕₕ(stable, z, L)
+    wₘ = calc_wₘ(stable, z, h, u_star, ϕₕ, w_star)
+    γc = (a*w_star*(w_C_0)/(wₘ^2*h)) * (z/h <= 0.1) * (stable >= 0)
+    return γc
+end
+
+"""
+When the PBL height is taken from the meteorological datasets 
+"""
+eqs = [
+    θ_s ~ PT(T_s,P_s)
+    θ_10 ~ PT(T_10,P_10)
+    θᵥ_s ~ VPT(θ_s,q_s)
+    θᵥ_10 ~ VPT(θ_10,q_s)
+    V_10 ~ (u_10^2+v_10^2)^0.5
+    stable ~ calc_stable(θᵥ_s,θᵥ_10)
+    w_u_0 ~ -V_10*u_10*Surface_fluxes(stable, V_10, u_10, v_10, θ_s, θ_10, q_s, q_10, θᵥ_s, θᵥ_10, z1, z_0m)[1]
+    w_v_0 ~ -V_10*v_10*Surface_fluxes(stable, V_10, u_10, v_10, θ_s, θ_10, q_s, q_10, θᵥ_s, θᵥ_10, z1, z_0m)[1]
+    w_θv_0 ~ V_10*(θᵥ_s-θᵥ_10)*Surface_fluxes(stable, V_10, u_10, v_10, θ_s, θ_10, q_s, q_10, θᵥ_s, θᵥ_10, z1, z_0m)[2]
+
+    Kc ~ calc_Kc(w_u_0, w_v_0, θᵥ_s, w_θv_0, θᵥ_10, z, h)
+    γc ~ calc_γc(θᵥ_s, θᵥ_10, z, h, w_θv_0, w_u_0, w_v_0, C(0,t), C(z1,t), z1)
+    Dt(C(z,t)) ~ Dz(-Kc*(Dz(C(z,t))-γc))
+]
+
+z_min = t_min = 0
+z_max = 200
+t_max = 10
+
+domains = [
+    z ∈ Interval(z_min,z_max)
+    t ∈ Interval(t_min, t_max)
+]
+
+# Periodic BCs
+bcs = [
+    C(z,0) ~ 10
+    C(0,t) ~ 10
+]
+
+@named pdesys = PDESystem(eqs,bcs,domains,[z,t],[C(z,t)],[u_10=>1.78, v_10=>2.53, T_s=>273.54, T_10=>275.51, q_s=>0.00377, q_10=>0.00377, z1=>10, z_0m=>0.368, h=>208, P_s=>1013.25, P_10=>1012.04])
+
+N = 10
+
+order = 2 # This may be increased to improve accuracy of some schemes
+
+# Integers for x and y are interpreted as number of points. Use a Float to directtly specify stepsizes dx and dy.
+discretization = MOLFiniteDifference([z=>N], t, approx_order=order)
```

Comment:
When I orginally tried to run this, I got the same weird error that you showed me. However, I got the same error when I ran [this tutorial](https://docs.sciml.ai/MethodOfLines/stable/tutorials/heat/#Dirichlet-boundary-conditions), so I guessed that maybe I was using an old version of the library. I tried to update MethodOfLines.jl but it didn't let me until I forced it to update by doing this:

``` julia
pkg> add MethodOfLines@v0.10.2
```
I think maybe it doesn't let you do that by default if updating the package in question would require downgrades in other packages.

Anyways, after upgrading MethodOfLines to the latest version, it works when a simplified version of the equation system:

``` julia
@constants w=1 [unit = u"m/s", description = "vertical velocity"]

eqs = [
    Dt(C(z,t)) ~ Dz(C(z,t)) * w
]
```

except that it runs into [this bug](https://github.com/SciML/MethodOfLines.jl/issues/287). 

You can get around that bug by adding this code:

``` julia
# Circumvent unit checks
ModelingToolkit.check_units(eqs...) = nothing
```
After doing that the simplified version works, but the full version still runs into some issue that seems to be related to `z1`. To debug it you can try starting with the simplified version that does work and then slowly adding more of the complexity in until it stops working, which should help make it clear what the issue is.

Reply:
Thank you for the useful suggestion! I will work on that!